Installation failure for `CMake` 3.16.3

Pulling the latest changes applied in #113 leads to a plenty of error messages with the first one being

```bash
CMake Error at applications/CMakeLists.txt:38 (ADD_SUBDIRECTORY):
  The source directory

    ~/exadg/applications/CMakeFiles

  does not contain a CMakeLists.txt file.
```

I followed the installation instruction provided in the wikis.

--- a/cmake/macros/macro_subdir_list.cmake
+++ b/cmake/macros/macro_subdir_list.cmake
@@ -25,12 +25,17 @@
 #
 #########################################################################
 
+# The following code is a modified version of the code proposed in:
+# https://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory
+
 MACRO(SUBDIRLIST result curdir)
   FILE(GLOB children RELATIVE ${curdir} ${curdir}/*)
   SET(dirlist "")
   FOREACH(child ${children})
     IF(IS_DIRECTORY ${curdir}/${child})
-      LIST(APPEND dirlist ${child})
+       IF(NOT "${child}" STREQUAL "CMakeFiles")
+         LIST(APPEND dirlist ${child})
+       ENDIF()
     ENDIF()
   ENDFOREACH()
   SET(${result} ${dirlist})

